pydantic/main.py

```diff
@@ -181,22 +181,26 @@ def __getstate__(self):
     def __setstate__(self, state):
         object.__setattr__(self, '__values__', state)
 
-    def dict(self, *, include: Set[str]=None, exclude: Set[str]=set()) -> Dict[str, Any]:
+    def dict(self, *, include: Set[str]=None, exclude: Set[str]=set(), by_alias: bool = False) -> Dict[str, Any]:
```

Comment:
I guess `by_alias` needs to be recursive (eg. work on submodels).

To do this:
* rename `__iter__()` to `_iter(by_alias=False)`
* `__iter__()` can then just call `_iter()`
* `_get_value` should have signature `_get_value(by_alias)` and then pass `by_alias` on.

Reply:
updated